src/common/rules.mk

```diff
@@ -58,7 +58,6 @@ src_common_libcommon_la_SOURCES += \
     src/common/platform.h \
     src/common/debugman.h \
     src/common/debug.h \
-    src/common/atomic.h \
```

Comment:
We can then also remove remove Boost.Atomic from
- our build files
  - https://github.com/xoreos/xoreos/blob/master/CMakeLists.txt#L264
  - https://github.com/xoreos/xoreos/blob/master/configure.ac#L345
  - https://github.com/xoreos/xoreos/blob/master/Makefile.am#L74
- our documentation
  - https://github.com/xoreos/xoreos/blob/master/BUILDDEP.md
- the distribution build files
  - https://github.com/xoreos/xoreos/blob/master/dists/fedora/xoreos.spec#L36
  - https://github.com/xoreos/xoreos/blob/master/dists/debian/control#L5